src/Components/Artwork/Details/PrimaryLabelLine.tsx

```diff
@@ -1,23 +1,35 @@
 import { Text } from "@artsy/palette"
 import { useArtworkGridContext } from "Components/ArtworkGrid/ArtworkGridContext"
-import { graphql, createFragmentContainer } from "react-relay"
-import { PrimaryLabelLine_artwork$data } from "__generated__/PrimaryLabelLine_artwork.graphql"
-import { FC } from "react"
 import { SystemQueryRenderer } from "System/Relay/SystemQueryRenderer"
-import { PrimaryLabelLineQuery } from "__generated__/PrimaryLabelLineQuery.graphql"
+import type { PrimaryLabelLineQuery } from "__generated__/PrimaryLabelLineQuery.graphql"
+import type { PrimaryLabelLine_artwork$data } from "__generated__/PrimaryLabelLine_artwork.graphql"
+import { type FC, useEffect } from "react"
+import { createFragmentContainer, graphql } from "react-relay"
 
 interface PrimaryLabelLineProps {
   label: string | null | undefined
   artwork?: PrimaryLabelLine_artwork$data
 }
 
-export const PrimaryLabelLine: React.FC<React.PropsWithChildren<PrimaryLabelLineProps>> = ({
-  label,
-  artwork,
-}) => {
-  const { hideSignals } = useArtworkGridContext()
+export const PrimaryLabelLine: React.FC<
+  React.PropsWithChildren<PrimaryLabelLineProps>
+> = ({ label, artwork }) => {
+  const { hideSignals, updateSignals } = useArtworkGridContext()
   const partnerOffer = artwork?.collectorSignals?.partnerOffer
 
+  useEffect(() => {
+    if (updateSignals && artwork) {
+      const signals: string[] = []
+      if (partnerOffer) {
+        signals.push("PARTNER_OFFER")
+      }
+      if (label) {
+        signals.push(label)
+      }
+      updateSignals(artwork.internalID, signals)
```

Comment:
This is the only place we update signals (for tracking purposes). I think it makes sense because this component represents the display of the collector signals to users. We might have to keep in mind if in the future the visual or the exact component implementation changes, we'll update the tracking accordingly.